src/publisher.ts

```diff
@@ -77,7 +101,14 @@ export class Publisher {
   private sayHello() {
     for (let iface of this.getInterfaces()) {
       const message = new Buffer(this.buildMessage(iface.address));
-      this.client.send(message, 0, message.length, PORT, iface.broadcast);
+
+      if (this.client) {
```

Comment:
I think if `this.client` is null at this point something really bad happened. We want it to crash or at least print some error. Not silently do nothing.

We could move this to the top of the function as an assert:
```
if (!this.client) {
   console.error('Bad internal state, Socket is undefined');
   return;
}
```